### PR TITLE
Adds Spring Integration Channel Adapters for GCP Pub/Sub code sample

### LINKED ISSUE #49: Add sample application for PubSub Channel Adapter
Maybe Storage Resource Abstraction as well.

### PR DESCRIPTION
**Pull request by joaoandremartins**
Adds a couple of applications, one that sends a message and another one
that receives and processes it by logging it.
Also slightly fixes other code samples.

Fixes #49 and #146 partially.

### CODE REVIEW

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+1. Configure your GCP project ID and credentials by following
+link:../../spring-cloud-gcp-starters/spring-cloud-gcp-starter-core[these instructions], or just let
+Spring discover those parameters for you.
+2. Go to the https://pantheon.corp.google.com/cloudpubsub/topicList[Google Cloud Console] and create
+a topic called `exampleTopic` and a subscription called `exampleSubscription`.
+3. Start the `spring-cloud-gcp-spring-integration-pubsub-example-sender` and
+`spring-cloud-gcp-spring-integration-pubsub-example-receiver` applications, which are configured to
+use ports 8081 and 8082.
+4. Go to http://localhost:8081, write a message and hit the `Submit` button.
+5. Verify that the receiver logged the message you wrote.
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc]: Need more instructions on how to verify that the message arrived.
What exactly should you see in the log?

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Pub/Sub topic.
+The other app subscribes to that topic and logs received messages.
+
+== Setup & Configuration
+
+1. Configure your GCP project ID and credentials by following
+link:../../spring-cloud-gcp-starters/spring-cloud-gcp-starter-core[these instructions], or just let
+Spring discover those parameters for you.
+2. Go to the https://pantheon.corp.google.com/cloudpubsub/topicList[Google Cloud Console] and create
+a topic called `exampleTopic` and a subscription called `exampleSubscription`.
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc]: "subscription **for the topic**" for clarity

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.channel.DirectChannel;
+import org.springframework.integration.gcp.pubsub.AckMode;
+import org.springframework.integration.gcp.pubsub.inbound.PubSubInboundChannelAdapter;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class WebController {
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java]: Why do you need a web controller, if you're not actually serving anything? In fact, you don't need the port or the web starter.

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-sender/src/main/java/com/example/WebController.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+ */
+@RestController
+public class WebController {
+
+	@Autowired
+	private PubsubOutboundGateway messagingGateway;
+
+	@Bean
+	@ServiceActivator(inputChannel = "pubsubOutputChannel")
+	public MessageHandler messageSender(PubSubOperations pubsubTemplate) {
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-sender/src/main/java/com/example/WebController.java]: This bean should probably be defined in a `@Configuration/@SpringBootApplication` class, not the controller.

File: spring-cloud-gcp-examples/spring-cloud-gcp-trace-example/pom.xml [hunk 5]
```diff
@@ -4,7 +4,8 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>com.example</groupId>
-    <artifactId>spring-gcp-trace-example</artifactId>
+    <artifactId>spring-cloud-gcp-trace-example</artifactId>
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-trace-example/pom.xml]: Seems orthogonal to this PR.

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+One app gets a message provided by the user through a webpage and publishes it to a Google Cloud
+Pub/Sub topic.
+The other app subscribes to that topic and logs received messages.
+
+== Setup & Configuration
+
+1. Configure your GCP project ID and credentials by following
+link:../../spring-cloud-gcp-starters/spring-cloud-gcp-starter-core[these instructions], or just let
+Spring discover those parameters for you.
+2. Go to the https://pantheon.corp.google.com/cloudpubsub/topicList[Google Cloud Console] and create
```
Comment (meltsufin) [inline spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc]: I skipped this step initially, and tried publishing a message anyway, and I saw no error in the logs or the web UI, yet no message went anywhere.

File: spring-cloud-gcp-examples/pom.xml [hunk 7]
```diff
@@ -21,5 +21,6 @@
     <modules>
 		<module>spring-cloud-gcp-config-example</module>
 		<module>spring-cloud-gcp-trace-example</module>
+		<module>spring-cloud-gcp-spring-integration-pubsub-example</module>
```
Comment (artembilan) [inline spring-cloud-gcp-examples/pom.xml]: Double `spring`. How about `spring-integration-cloud-gcp-pubsub-example` ? 
Or just without the second `spring`, or just `pubsub`, without `spring-integration`

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java (hunk 3 above)
Comment (artembilan) [reply spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java]: Agree. This application does nothing with Web. Let's get rid of it here!

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+				new PubSubInboundChannelAdapter(pubSubTemplate, "exampleSubscription");
+		adapter.setOutputChannel(inputChannel);
+		adapter.setAckMode(AckMode.MANUAL);
+
+		return adapter;
+	}
+
+	@Bean
+	@ServiceActivator(inputChannel = "pubsubInputChannel")
+	public MessageHandler messageReceiver() {
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java]: Would be great to demonstrate some other Spring Integration tricks like:
```
@ServiceActivator(inputChannel = "pubsubInputChannel")
public void messageReceiver(String payload,
                        @Header(GcpHeaders.ACKNOWLEDGEMENT) AckReplyConsumer consumer) {
   ...
}
```
and no need `@Bean` annotation as well.

File: spring-cloud-gcp-examples/spring-cloud-gcp-trace-example/pom.xml (hunk 5 above)
Comment (artembilan) [reply spring-cloud-gcp-examples/spring-cloud-gcp-trace-example/pom.xml]: But still it's simple correct fix.
Let it be!

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc (hunk 6 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc]: That happens because the call is made asynchronously.
Since #166 was merged, you can set the mode to synchronous and the error will show up. We mention in the README and ref doc that on async mode, errors won't be displayed.

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java (hunk 8 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java]: Hmm interesting that `messageReceiver` doesn't require `@Bean` but `messageSender` does!

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc (hunk 6 above)
Comment (joaoandremartins) [reply spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/README.adoc]: Actually, for the example, I'll set sync to `true` so errors become visible!

File: spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java (hunk 8 above)
Comment (artembilan) [reply spring-cloud-gcp-examples/spring-cloud-gcp-spring-integration-pubsub-example/spring-cloud-gcp-spring-integration-pubsub-example-receiver/src/main/java/com/example/WebController.java]: Because `messageSender` definitely represents bean - real `MessageProducer`, but `messageReceiver()` is already just a POJO method: https://docs.spring.io/spring-integration/docs/4.3.12.RELEASE/reference/html/overview.html#programming-tips

File: spring-cloud-gcp-examples/spring-cloud-gcp-integration-pubsub-example/spring-cloud-gcp-integration-pubsub-example-receiver/src/main/resources/application.properties [hunk 9]
```diff
@@ -0,0 +1 @@
+server.port=8082
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-integration-pubsub-example/spring-cloud-gcp-integration-pubsub-example-receiver/src/main/resources/application.properties]: You don't have Web in this app any more. Therefore no any `server.port` properties.
I would say no `application.properties` at all!

File: spring-cloud-gcp-examples/spring-cloud-gcp-integration-pubsub-example/spring-cloud-gcp-integration-pubsub-example-sender/src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1 @@
+server.port=8081
```
Comment (artembilan) [inline spring-cloud-gcp-examples/spring-cloud-gcp-integration-pubsub-example/spring-cloud-gcp-integration-pubsub-example-sender/src/main/resources/application.properties]: This is stand along sample - I would prefer to let end-users to get a gain from using just default 8080 port.
Therefore no this `application.properties` at all!
